src/components/MultiSelect/MultiSelectItemsCountButton.tsx

```diff
@@ -0,0 +1,87 @@
+import React, { forwardRef } from "react";
+import { useMultiStyleConfig } from "@chakra-ui/react";
+import Button from "../Button/Button";
+import Icon from "../Icons/Icon";
+import { SelectedItems } from "./MultiSelect";
+
+export interface MultiSelectItemsCountButtonProps {
+  id: string;
+  /** The id of the MultiSelect using this button. */
+  multiSelectId: string;
+  /** The label text rendered within the MultiSelect using this button. */
+  multiSelectLabelText: string;
+  /** The open status of the MultiSelect menu. */
+  isOpen: boolean;
+  /** The selected items state (items that were checked by user). */
+  selectedItems: SelectedItems;
+  /** The callback function for the menu toggle. */
+  onMenuToggle?: () => void;
+  /** The action to perform for clear/reset button of MultiSelect. */
+  onClear?: () => void;
+  onKeyDown?: () => void;
+  /** Ref to the Accordion Button element. */
+  accordianButtonRef: any;
+}
+
+/**
+ * The toggle button component used to open and close the `MultiSelect` menu.
+ * A second button is rendered above the main button that displays the current
+ * number of selected items. Clicking on the second button will clear all
+ * the selected items and the main button's close event will not be fired
+ * (as expected).
+ */
+const MultiSelectItemsCountButton = forwardRef<
+  HTMLButtonElement,
+  MultiSelectItemsCountButtonProps
+>((props, _ref) => {
+  const {
+    isOpen,
+    multiSelectId,
+    multiSelectLabelText,
+    accordianButtonRef,
+    onClear,
+    selectedItems,
+  } = props;
+
+  // Sets the selected items count on the menu button.
+  let getSelectedItemsCount;
+  let selectedItemsAriaLabel;
+  if (selectedItems[multiSelectId]?.items.length > 0) {
+    getSelectedItemsCount = `${selectedItems[multiSelectId].items.length}`;
+    const itemPlural = getSelectedItemsCount === "1" ? "" : "s";
+    selectedItemsAriaLabel = `remove ${getSelectedItemsCount} item${itemPlural} selected from ${multiSelectLabelText}`;
+  }
+
+  const styles = useMultiStyleConfig("MultiSelectItemsCountButton", {
+    isOpen,
+    hasSelectedItems: getSelectedItemsCount,
```

Comment:
Replace all the `getSelectedItemsCount` with the new passed prop `selectedItemsCount`.